### ISSUE TITLE
Make if/then/else rebindable syntax

### ISSUE THREAD

**Issue by rightfold** [purescript/purescript#3164]
Such that `if e then f else g` would desugar to `ifThenElse e (\_ -> f) (\_ -> g)`. Then, add an inliner rule for `Prelude.ifThenElse`.

**Comment by paf31** [comment]
I agree this would be useful, I've wanted it a few times. It would need a new inliner rule too.

**Comment by chexxor** [comment]
Why is if-then-else a special syntax? I wonder if we could just move it to a library. Could change it to use JS ternary, then, like `a ? b : c`. Has anyone explored the idea of "n-ary" operators? We have binary operators, which allow making two-argument infix operators. Having an operator delimit each argument to the function would require changing the parser to support infixBracketed (could call it) expressions, I guess.

``` purescript
module Boolean where
ifThenElse :: Boolean -> a -> b
infixBracketed ifThenElse as ?,:

module ThreeValueLogic where
ifThenElse :: TripleBoolean -> a -> b
infixBracketed ifThenElse as ?,:

module Example where
import Boolean ((?,:))
-- or import ThreeValueLogic
let a = boolLike ? consequenceFn a b c : alternateFn d e f
```

**Comment by natefaubion** [comment]
@chexxor n-ary operators is called "mixfix", which you'll see in languages like Agda https://github.com/purescript/purescript/issues/139

**Comment by hdgarrood** [comment]
fun fact, you can have that C-style ternary already with regular old binary operators: https://twitter.com/hdgarrood/status/618264003714936832

**Comment by hdgarrood** [comment]
There's definitely an argument to be made for removing if-then-else entirely, but in my view it's not strong enough to be worth seriously considering for PureScript at this point, since it would break so much code.
